lib/json/json.go

```diff
@@ -100,8 +102,19 @@ func encode(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, k
 		}
 	}
 
+	path := make([]unsafe.Pointer, 0, 8)
+
 	var emit func(x starlark.Value) error
 	emit = func(x starlark.Value) error {
+		if ptr := pointer(x); ptr != nil {
```

Comment:
// It is only necessary to push/pop the item when it might contain
// itself (i.e. the last three switch cases), but omitting it in the other
// cases did not show significant improvement on the benchmarks.